### ISSUE TITLE
ChirpStack 4.7 breaks sslmode require

### ISSUE THREAD

**Issue by trivialkettle** [chirpstack/chirpstack#387]
Hi,
I run ChirpStack in docker compose and run a gcloud SQL instance.
Until version 4.6 I could connect to it with `sslmode=require`

After the update to 4.7 I get the following error:
```
Error: Error occurred while creating a new object: error performing TLS handshake: invalid peer certificate: UnknownIssuer
Caused by:
error performing TLS handshake: invalid peer certificate: UnknownIssuer
```

I am not sure what option I should set to `ca_cert`, since `sslmode=require` should skip the verification.

**Comment by brocaar** [comment]
Rustls is more strict, for this reason you must configure the ca_cert if it
is not already configured on the host.

You should be able to download the server certificate from the gcp console.

**Comment by pyttel** [comment]
> Hi again, I could need some assistance in setting up the psql dsn for a gcloud sql instance
> 
> I tried e.g. (also with `/` substituted by `%2F`)
> 
> ```
> dsn="postgres://dbname?host=/cloudsql/instance:zone:db/.s.PGSQL.5432&user=user:passwort&sslmode=require
> ```
> 
> and get the following error:
> 
> ```
> Error: Error occurred while creating a new object: error connecting to server: No such file or directory (os error 2)
> ```
> 
> Edit: previously I used the private IP to connect, but this throws
> 
> ```
> Error: Error occurred while creating a new object: error performing TLS handshake: invalid peer certificate: NotValidForName
> ```

We face the same issue on kubernetes cluster. We tried a lot of stuff. But we are still getting the error. With a custom build and a panic included we get the following exception since v4.7.0. Version 4.6 is working fine.

Logs from deadpool dependency do not include much information. So, we are still analyzing.

![grafik](https://github.com/chirpstack/chirpstack/assets/80945247/4da27123-1ad5-4a66-8975-444b3bad27e3)

**Comment by pyttel** [comment]
We found the issue: We used the env variable SSL_CERT_FILE which pointed to a certificate file which did not exist. Because of that we must use postgres.ca_cert from config to overwrite the env variable SSL_CERT_FILE. Using valid values or not using any of above options solved our problem. So, it was our fault. Hopefully, our experience helps somebody and saves some hours^^
